js_modules/dagit/packages/core/src/asset-graph/AssetNode.tsx

```diff
@@ -318,46 +345,126 @@ export const ComputeStatusNotice: React.FC<{computeStatus: AssetComputeStatus}>
-        <Spinner purpose="body-text" stopped />
-      </Tooltip>
-      <AssetRunLink runId={unstartedRunIds[0]} />
-    </Box>
-  ) : runWhichFailedToMaterialize?.__typename === 'Run' ? (
-    <Box flex={{gap: 4, alignItems: 'center'}}>
-      <Tooltip
-        content={`Run ${titleForRun({
-          runId: runWhichFailedToMaterialize.id,
-        })} failed to materialize this asset`}
-      >
-        <Icon name="warning" color={Colors.Red500} />
-      </Tooltip>
-      <AssetRunLink runId={runWhichFailedToMaterialize.id} />
+  const runTagContent =
+    inProgressRunIds?.length > 0 ? (
+      <Box flex={{gap: 4, alignItems: 'center'}}>
+        <Tooltip content="A run is currently rematerializing this asset.">
+          <Spinner purpose="caption-text" />
+        </Tooltip>
+        <AssetRunLink runId={inProgressRunIds[0]} />
+      </Box>
+    ) : unstartedRunIds?.length > 0 ? (
+      <Box flex={{gap: 4, alignItems: 'center'}}>
+        <Tooltip content="A run has started that will rematerialize this asset soon.">
+          <Spinner purpose="caption-text" stopped />
+        </Tooltip>
+        <AssetRunLink runId={unstartedRunIds[0]} />
+      </Box>
+    ) : runWhichFailedToMaterialize?.__typename === 'Run' ? (
+      <Box flex={{gap: 4, alignItems: 'center'}}>
+        <Tooltip
+          content={`Run ${titleForRun({
+            runId: runWhichFailedToMaterialize.id,
+          })} failed to materialize this asset`}
+        >
+          <Icon name="warning" color={Colors.Red500} />
+        </Tooltip>
+        <AssetRunLink runId={runWhichFailedToMaterialize.id} />
+      </Box>
+    ) : lastMaterialization ? (
+      <Box flex={{gap: 6, alignItems: 'center'}}>
+        {lastMaterializationRunStatus ? (
+          <RunStatusIndicator status={lastMaterializationRunStatus} size={10} />
+        ) : (
+          <div style={{width: 10}} />
+        )}
+        <AssetRunLink
+          runId={lastMaterialization.runId}
+          event={{stepKey, timestamp: lastMaterialization.timestamp}}
+        />
+      </Box>
+    ) : undefined;
+
+  if (!includeFreshness) {
+    return runTagContent || <span>–</span>;
+  }
+
+  return (
+    <Box flex={{direction: 'row', gap: 4}}>
+      {runTagContent ? <Tag>{runTagContent}</Tag> : <span>–</span>}
+      {liveData && <CurrentMinutesLateTag liveData={liveData} policyOnHover />}
     </Box>
-  ) : lastMaterialization ? (
-    <AssetRunLink
-      runId={lastMaterialization.runId}
-      event={{stepKey, timestamp: lastMaterialization.timestamp}}
-    />
+  );
+};
+
+const lateness = (minutes: number) => {
+  const [val, unit] =
+    minutes > 60 * 24 * 60
+      ? [Math.ceil(minutes / (30 * 24 * 60)), 'month']
+      : minutes > 3 * 24 * 60
+      ? [Math.ceil(minutes / (24 * 60)), 'day']
+      : minutes > 60
+      ? [Math.ceil(minutes / 60), 'hour']
+      : [Math.ceil(minutes), 'min'];
+
+  return val !== 1 && unit !== 'min' ? `${val} ${unit}s` : `${val} ${unit}`;
+};
```

Comment:
Can this use `moment` or something so that we don't need to implement it ourselves?

Reply:
Ahh let me see! There are a couple caveats to this one (showing up to 3 days as hours) but we can do away with that and use moment i think